src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -1587,6 +1587,38 @@ namespace winrt::TerminalApp::implementation
         });
     }
 
+    // Method Description:
+    // - Set read-only mode on the active pane
+    // - If a parent pane is selected, this will ensure that all children have
+    //   the same read-only status.
+    void TerminalTab::SetPaneReadOnly(const bool readOnlyState)
+    {
+        auto hasReadOnly = false;
+        auto allReadOnly = true;
+        _activePane->WalkTree([&](const auto& p) {
+            if (const auto& control{ p->GetTerminalControl() })
+            {
+                hasReadOnly |= control.ReadOnly();
+                allReadOnly &= control.ReadOnly();
+            }
+        });
+        _activePane->WalkTree([&](const auto& p) {
+            if (const auto& control{ p->GetTerminalControl() })
+            {
+                // If all controls have the same read only state then just disable
+                if (allReadOnly || !hasReadOnly)
+                {
+                    control.SetReadOnly(readOnlyState);
+                }
+                // otherwise set to all read only.
+                else if (!control.ReadOnly())
+                {
+                    control.SetReadOnly(readOnlyState);
+                }
+            }
+        });
+    }
+

```

Comment:
Is all of this logic necessary? Why wouldn't this just be a 

```c++
WalkTree([&](const auto& p){ 
    if (const auto& control{ p->GetTerminalControl() })
    {
        control.SetReadOnly(readOnlyState);
    }
});
```

Maybe I just haven't had enough coffee yet

Reply:
Ill push my changes for this, I dont see any issue when ive tested it.